src/content/concepts/module-federation.md

```diff
@@ -120,3 +125,101 @@ Each page of a Single Page Application is exposed from container build in a sepa
+
+## Troubleshooting
+
+`Uncaught Error: Shared module is not available for eager consumption` <br/> The application is eagerly executing an app which is operating as an omnidirectional host. There are options to choose from: <br> You can set the dependency as eager inside the advanced API of Module Federation, which doesn’t put the modules in an async chunk, but provides them synchronously.<br>This allows us to use these shared modules in the initial chunk. But be careful as all provided and fallback modules will always be downloaded. <br> It’s wise to provide it only at one point of your app, e.g. the shell.
+
+
+We strongly recommend using an async boundary. Doing so will not create any additional round trips, it’s also more performant in general as initialization code is split out of a larger chunk.
+`ìmport('./bootstrap')`
+
+Create a `bootstrap.js` file, and move the original contents of the entry point code into that file.
+
+```js
+//index.js
+import('./bootstrap');
+//bootstrap.js
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+ReactDOM.render(<App />, document.getElementById('root'));
+```
+
+Methods mentioned above work, but can have some limits or drawbacks.
+
+```js
+new ModuleFederationPlugin({
+  shared: {
+    ...deps,
+    react: {
+      eager: true,
+    }
+  }
+});
+```
+
+`bundle-loader` can be used as an alternative to setting dependencies as "eager". This method is less performant as it will introduce additional round trips.
+
+```js
+const config = {
+  entry: 'bundle-loader!./bootstrap.js'
+};
+```
+
+Or you can set via module rules: [See Full Example](https://github.com/module-federation/module-federation-examples/blob/master/basic-host-remote/app1/webpack.config.js)
+
+```js
+const config = {
+  module: {
+    rules: [
+      {
+        test: /bootstrap\.js$/,
+        loader: 'bundle-loader',
+        options: {
+          lazy: true,
+        },
+      },
+    ]
+  }
+};
+```
+
+Then change the entry point to look like this.
+Create a `bootstrap.js` file, and move the original contents of the entry point code into that file.
+
+```js
+//index.js
+import bootstrap from './bootstrap';
+bootstrap();
+//bootstrap.js
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+ReactDOM.render(<App />, document.getElementById('root'));
+```
+
+`Uncaught Error: Module “./Button” does not exist in container.` <br> It likely does not say button, but the error message will look similar. <br> This issue is typically seen if you are upgrading from beta.16 to beta.17 <br> Within ModuleFederationPlugin. Change the exposes from:
+
+```js
+new ModuleFederationPlugin({
+  exposes: {
+    'Button': './src/Button'
+  }
+});
+```
```

Comment:
Use `diff` for convenience
```suggestion
```diff
new ModuleFederationPlugin({
  exposes: {
-   'Button': './src/Button'
+   './Button':'./src/Button'
  }
});
```
```